src/main-client.ts

```diff
@@ -600,6 +952,93 @@ export class MainClient extends BaseRestClient {
     return this.postPrivate('sapi/v1/managed-subaccount/withdraw', params);
   }
 
+  getManagedSubAccountSnapshot(
+    params: ManagedSubAccountSnapshotParams,
+  ): Promise<ManagedSubAccountSnapshotResponse> {
+    return this.getPrivate(
+      'sapi/v1/managed-subaccount/accountSnapshot',
+      params,
+    );
+  }
+
+  getManagedSubAccountTransfersInvestor(
+    params: ManagedSubAccountTransferLogParams,
+  ): Promise<ManagedSubAccountTransferLogResponse> {
+    return this.getPrivate(
+      'sapi/v1/managed-subaccount/queryTransLogForInvestor',
+      params,
+    );
+  }
+
+  getManagedSubAccountTransfersParent(
+    params: ManagedSubAccountTransferLogParams,
+  ): Promise<ManagedSubAccountTransferLogResponse> {
+    return this.getPrivate(
+      'sapi/v1/managed-subaccount/queryTransLogForTradeParent',
+      params,
+    );
+  }
+
+  getManagedSubAccountFuturesAssets(params: {
+    email: string;
+  }): Promise<ManagedSubAccountFuturesAssetsResponse> {
+    return this.getPrivate(
+      'sapi/v1/managed-subaccount/fetch-future-asset',
+      params,
+    );
+  }
+
+  getManagedSubAccountMarginAssets(params: {
+    email: string;
+  }): Promise<ManagedSubAccountMarginAssetsResponse> {
+    return this.getPrivate('sapi/v1/managed-subaccount/marginAsset', params);
+  }
+
+  getSubAccountAssetsMaster(params: {
+    email: string;
+  }): Promise<SubAccountAssetsMasterResponse> {
```

Comment:
```suggestion
  }): Promise<{ balances: whateverthetypeis[]; }> {
```

flatten these kind of types please